Source/objects.cpp

```diff
@@ -1074,8 +1074,15 @@ void AddSarc(int i)
 	dObject[Objects[i].position.x][Objects[i].position.y - 1] = -(i + 1);
 	Objects[i]._oVar1 = GenerateRnd(10);
 	Objects[i]._oRndSeed = AdvanceRndSeed();
-	if (Objects[i]._oVar1 >= 8)
-		Objects[i]._oVar2 = PreSpawnSkeleton()->getId();
+	if (Objects[i]._oVar1 >= 8) {
+		Monster *monster = PreSpawnSkeleton();
+		if (monster != nullptr) {
+			Objects[i]._oVar2 =	monster->getId();
+		} else {
+			Objects[i]._oVar1 = 0;

```

Comment:
0 is a valid monster ID (though it'll never be used in practice). `-1` like the function previously returned on failure is better.

Reply:
```suggestion
			Objects[i]._oVar1 = -1;
```